Library/Homebrew/utils/popen.rb

```diff
@@ -21,4 +24,44 @@ def self.popen(args, mode)
       end
     end
   end
+
+  def self.socket_popen(suspicious_fail_msg, &block)
```

Comment:
@mikemcquaid I have already avoid some copy and paste by adding this new method. However, you may have some comments on the method name and arguments design.